tidy3d/components/tcad/materials/charge.py

```diff
@@ -0,0 +1,97 @@
+from typing import Union
+
+import pydantic.v1 as pd
+
+from tidy3d.components.data.data_array import SpatialDataArray
+from tidy3d.components.tcad.bandgap import SlotboomNarrowingBandGap
+from tidy3d.components.tcad.generation_recombination import (
+    AugerRecombination,
+    RadiativeRecombination,
+    ShockleyReedHallRecombination,
+)
+from tidy3d.components.tcad.materials.abstract import AbstractHeatChargeSpec
+from tidy3d.components.tcad.mobility import CaugheyThomasMobility
+from tidy3d.components.tcad.types import (
+    BandGapModelTypes,
+    MobilityModelTypes,
+    RecombinationModelTypes,
+)
+from tidy3d.constants import ELECTRON_VOLT
+
+
+class ElectronicSpec(AbstractHeatChargeSpec):
+    # TODO does this account for the whole temporal dynamics? If not then it should be called ElectroStatic, which it was called before, but unsure why it got changed?
+    """
+    This class is used to define electro-static semiconductors.
```

Comment:
This was called `SemiconductorSpec` because it only makes sense for semiconductors. Temperature is included, not only in the different models but, more importantly, in the drift-diffusion equations, which consider Boltzmann statistics

Reply:
Yeah that makes sense.

The issue I have with the definition of SemiConductor is that this does not imply the "active" nature of a material. Si, Si02 are all semiconductors, but they can also be treated as insulators, or even good conductors, depending on the simulation conditions ie optical/electrical. Electronic is probably not the answer you are right. "ActiveSemiConductorMedium" maybe sounds good to me.

Maybe we can chat about this in https://github.com/flexcompute/tidy3d/pull/2124